cdap-api-spark-base/src/main/java/co/cask/cdap/api/spark/JavaSparkExecutionContextBase.java

```diff
@@ -110,6 +112,14 @@
    */
   public abstract MessagingContext getMessagingContext();
 
+  /**
+   * Returns a {@link MetadataReaderContext} which can be used to read metadata. Currently the returned instance can
+   * only be used in the Spark driver process.
+   *
+   * @return A {@link MetadataReaderContext}
+   */
+  public abstract MetadataReaderContext getMetadataReaderContext();
```

Comment:
is this a user-facing API? Feels like it should just be called getMatadataReader(). Actually, this context implements the MetadataReaderContext. So why is there a need for a public getMetadataReaderContext?

Reply:
Removed. See comment on another similar one.